doc/feature/lineEvents.md

```diff
@@ -0,0 +1,59 @@
+Line events
+===========
+
+At the moment, Fivox works only with point events, which is not enough in the
+cases where more precision is needed, e.g. LFP computation on neuron structures
+(line-shaped). We need to extend the functionality in order to support more
+types of events, such as lines.
+
+## Requirements
+
+* In addition to the current point events (we want to keep them to represent for
+  punctual structures), we need to add a new type of events, such as lines, to
+  represent elongated structures.
+* A line event must be represented by: two 3D points, a radius and a value.
+* We need the new events to be compatible with the existing ones. For example,
+  we could evaluate different types of events in the same use case, e.g. somas
+  (points) and dendrites (lines/cylinders).
+
+## Implementation
+
+The current Event class could serve as a base class and two new classes can
+derive from it: PointEvent and LineEvent. The base class will hold the common
+parts, such as the value of the event.
+
+Then, each loader will create their own events depending on the use case
+(points and/or lines), and the functors will evaluate all the events as if they
+were base events.
+The specific behaviour for each of the derived events can be implemented using:
+
+* Templated functions in the functor
+* Checking the class type through dynamic casting
+* Adding methods in the event class that will be implemented in the derived ones
+  (lineEvent, pointEvent, ...), doing whatever is needed depending on the type
+* ... more alternatives to be discussed
+
+## Issues
+
+### 1: Use only one value for the radius or two different values (beginning and
+end of line)?
+
+Resolved: Use only one radius
+
+We can use a single radius for the whole line, resulting in a cylinder instead
+of a conic structure. Using two different radii would only increase the
```

Comment:
That is up to the different loaders, same as now with the point events. The event defines only the semantics, but depending on the use case, it might be interpreted one way or another by the loader. For example, I would say that in the case of compartments, a good first approach is to use the average of the two ends' radii, but we could also use directly the radius at the beginning.
In any case, that's part of another discussion, this is just to discuss the convenience of using only one radius, after considering the implications.